subscription_manager/tests/test_controller_get_all.py

```diff
@@ -0,0 +1,118 @@
+        },
+        {
+            "comment": "Generation date: 23/06/2020, 10:46:41",
+            "currency": "CNY",
+            "frequency": "monthly",
+            "name": "Sky Store",
+            "owner": "Mary",
+            "price": 12.97,
+            "start_date": datetime(2019, 4, 13, 0, 0),
+        },
+        {
+            "comment": "Generation date: 23/06/2020, 10:47:36",
+            "currency": "EUR",
+            "frequency": "weekly",
+            "name": "Spotify",
+            "owner": "Robin",
+            "price": 30.23,
+            "start_date": datetime(2019, 10, 19, 0, 0),
+        },
+    ]
+
+
+subscriptions_obj_list = [
+    Subscription(
+        owner="Kevin",
+        name="Youtube Music",
+        frequency="daily",
+        start_date=date(2019, 5, 11),
+        price=8.3,
+        currency="USD",
+        comment="Generation date: 23/06/2020, 10:46:11",
+    ),
+    Subscription(
+        owner="Mary",
+        name="Sky Store",
+        frequency="monthly",
+        start_date=date(2019, 4, 13),
+        price=12.97,
+        currency="CNY",
+        comment="Generation date: 23/06/2020, 10:46:41",
+    ),
+    Subscription(
+        owner="Robin",
+        name="Spotify",
+        frequency="weekly",
+        start_date=date(2019, 10, 19),
+        price=30.23,
+        currency="EUR",
+        comment="Generation date: 23/06/2020, 10:47:36",
+    ),
+]
+
+
+@pytest.fixture
+def mock_dbhelper(subscriptions_db_list) -> DBHelper:
+    """Returns mock for DBHelper for get_subscription by name from database"""
+    mock = DBHelper
+    mock.get_all_subscriptions = MagicMock(return_value=subscriptions_db_list)
+    return mock
+
+
+@pytest.fixture
+def controller(mock_dbhelper: DBHelper) -> Controller:
+    """Returns Controller class instance"""
+    database = mock_dbhelper
+    return Controller(database)
+
+
+@pytest.mark.parametrize("owner", ["Lena", "", None])
+def test_get_all_subscriptions(controller: Controller, owner: str):
+    """Check that Controller returned correct list of Subscription objects:
+        - No _id specified in each subscription
+        - type is not dict, but Subscription
+        - start_date field type is not datetime, but date
+    """
+    result = controller.get_subscriptions_list(owner)
+    assert result == subscriptions_obj_list
+
+
+@pytest.mark.parametrize("owner", [123, ("Lena", "Artur"), {"owner": "Lena"}])
+def test_get_all_subscriptions_wrong_type(controller: Controller, owner):
+    """Check that Controller raises exception in case of invalid owner field type"""
+    with pytest.raises(SubscriptionException) as exc:
+        result = controller.get_subscriptions_list(owner)
+        assert result == subscriptions_obj_list
+    expected_exc_msg = f"Found wrong owner name's type, expected: <str>, received: {type(owner), owner}"
+    assert expected_exc_msg == str(exc.value)
+
+
+def test_get_next_payment_date(controller: Controller):
```

Comment:
we should validate next_payment_date vale

Reply:
Will be done in another pull request in Subscription class tests.
Here I check that client is able to receive next payment date from the list of subscriptions